File: sdk/php/src/Resources/Services.php
<?php

namespace HiTechCloud\SDK\Resources;

class Services extends BaseResource
{
    /**
     * List services
     */
    public function list(array $params = []): array
    {
        return $this->http->get('/api/service', $params);
    }

    /**
     * Get service
     */
    public function get(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId);
    }

    /**
     * Get service methods
     */
    public function getMethods(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/methods');
    }

    /**
     * Cancel service
     */
    public function cancel(int $serviceId, array $data = []): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/cancel', $data);
    }

    /**
     * Get service label
     */
    public function getLabel(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/label');
    }

    /**
     * Set service label
     */
    public function setLabel(int $serviceId, array $data): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/label', $data);
    }

    /**
     * Renew service
     */
    public function renew(int $serviceId): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/renew');
    }

    /**
     * Get service cycle
     */
    public function getCycle(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/cycle');
    }

    /**
     * Set service cycle
     */
    public function setCycle(int $serviceId, array $data): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/cycle', $data);
    }

}

File: sdk/csharp/Resources/Services.cs
using System.Collections.Generic;
using System.Threading.Tasks;

namespace HiTechCloud.SDK.Resources;

public class Services : BaseResource
{
    public Services(HttpClientWrapper http) : base(http) { }

    public async Task<Dictionary<string, object?>> ServicesAsync()
    {
        return await Http.GetAsync("/api/service");
    }

    public async Task<Dictionary<string, object?>> ServiceAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}", id));
    }

    public async Task<Dictionary<string, object?>> MethodsAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/methods", id));
    }

    public async Task<Dictionary<string, object?>> CancelAsync(int id, Dictionary<string, object> data)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/cancel", id), data);
    }

    public async Task<Dictionary<string, object?>> LabelAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/label", id));
    }

    public async Task<Dictionary<string, object?>> SetLabelAsync(int id, Dictionary<string, object> data)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/label", id), data);
    }

    public async Task<Dictionary<string, object?>> RenewAsync(int id, Dictionary<string, object> data)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/renew", id), data);
    }

    public async Task<Dictionary<string, object?>> CycleAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/cycle", id));
    }

    public async Task<Dictionary<string, object?>> SetCycleAsync(int id, Dictionary<string, object> data)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/cycle", id), data);
    }
}
